nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java

```diff
@@ -0,0 +1,273 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard;
+
+import org.apache.nifi.annotation.behavior.DefaultRunDuration;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.SideEffectFree;
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@SideEffectFree
+@SupportsBatching(defaultDuration = DefaultRunDuration.TWENTY_FIVE_MILLIS)
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"attributes", "modification", "filter", "retain", "remove", "delete", "regex", "regular expression", "Attribute Expression Language"})
+@CapabilityDescription("Filters the attributes of a FlowFile by retaining specified attributes and removing the rest or by removing specified attributes and retaining the rest.")
+public class FilterAttribute extends AbstractProcessor {
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .description("All successful FlowFiles are routed to this relationship").name("success").build();
+
+    private final static Set<Relationship> relationships = Collections.singleton(REL_SUCCESS);
+
+
+    public static final AllowableValue FILTER_MODE_VALUE_RETAIN = new AllowableValue(
+            "RETAIN",
+            "Retain",
+            "Retains only the attributes matching the filter, all other attributes are removed."
+    );
+
+    public static final AllowableValue FILTER_MODE_VALUE_REMOVE = new AllowableValue(
+            "REMOVE",
+            "Remove",
+            "Removes the attributes matching the filter, all other attributes are retained."
+    );
+
+    public static final PropertyDescriptor FILTER_MODE = new PropertyDescriptor.Builder()
+            .name("FILTER_MODE")
+            .displayName("Filter mode")
```

Comment:
The `name` value should be the same as the displayName for new properties, making it easier for programmatic and UI access.
```suggestion
            .name("Filter Mode")
            .displayName("Filter Mode")
```

Reply:
That's true, sorry for not flagging this earlier. We should make the change on the main branch as well, then we can make sure both are aligned for the next releases.